### PR TITLE
Fix entity naming mechanism failure in MAL

### PR DESCRIPTION
**Pull request by wu-sheng**
- [ ] If this pull request closes/resolves/fixes an existing issue, replace the issue number. Closes #<issue number>.
- [x] Update the [`CHANGES` log](https://github.com/apache/skywalking/blob/master/CHANGES.md).

Fix entity(service/instance/endpoint) names in the MAL system(prometheus, native meter, open census, envoy metric service) are not controlled by core's naming-control mechanism.

### CODE REVIEW

**Review by hanahmily — APPROVED** [review]
LGTM

File: oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/analysis/meter/MeterEntity.java [hunk 1]
```diff
@@ -35,6 +36,8 @@
 @Getter
 @Builder(toBuilder = true)
 public class MeterEntity {
+    private static NamingControl NAMING_CONTROL;
```
Comment (kezhenxu94) [inline oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/analysis/meter/MeterEntity.java:39]: Where is it used?

Comment (wu-sheng) [reply oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/analysis/meter/MeterEntity.java:39]: I forgot to submit the second change. Just did.

**Review by kezhenxu94 — CHANGES_REQUESTED** [review]
This fix is fragile, `MeterEntity` exposes many means to construct its instances, this patch only covers the static methods `newService`, `newServiceInstance`, `newEndpoint`, `newServiceRelation`, future developers who use the uncovered cases to construct the entity might cause the bug again.

The other uncovered cases:

1. `MeterEntity.builder().serviceName("").instanceName("")...`, if it's designed to only use the `newXxxx` methods to construct `MeterEntity`, we should update the `@Builder` annotation to `@Builder(toBuilder = true, access = AccessLevel.PRIVATE)`
2. `meterEntity.toBuilder()` returns another builder, which is the same case with (1.), we'd better also remove `toBuilder = true` in `@Builder`
3. `new` constructor (although it's package-private), we'd better also make it `private`.

**Comment by wu-sheng** [comment]
@kezhenxu94 I removed `@Builder` and fixed the tests.

**Review by kezhenxu94 — APPROVED** [review]

**Pull request merged**